gateway-core/components/org.wso2.carbon.gateway.core/src/test/java/org/wso2/carbon/gateway/core/config/WUMLListenerTest.java

```diff
@@ -41,17 +46,108 @@ public void setup() {
+
+        Integration sampleIntegration = IntegrationConfigRegistry.getInstance().getIntegrationConfig(integrationName);
+        MediatorCollection generatedMediatorCollection = sampleIntegration.getResource(resourceName).getDefaultWorker()
+                .getMediators();
+        List<Mediator> mediatorList = generatedMediatorCollection.getMediators();
+
+        // Evaluate the generated mediator collection
+        if (mediatorList.size() == 6) {
+            asExpected = (generatedMediatorCollection.getFirstMediator() instanceof PropertyMediator);
+            asExpected = (mediatorList.get(1) instanceof LogMediator) && asExpected;
+            asExpected = (mediatorList.get(2) instanceof FilterMediator) && asExpected;
+            asExpected = (mediatorList.get(3) instanceof LogMediator) && asExpected;
+            asExpected = (mediatorList.get(4) instanceof CallMediator) && asExpected;
+            asExpected = (mediatorList.get(5) instanceof RespondMediator) && asExpected;
+
+            // expand the filter mediator
+            List<Mediator> thenMediatorList = ((FilterMediator) mediatorList.get(2)).getChildThenMediatorList()
+                    .getMediators();
+            List<Mediator> elseMediatorList = ((FilterMediator) mediatorList.get(2)).getChildOtherwiseMediatorList()
+                    .getMediators();
+
+            // expand the filter mediator inside if-then block
+            asExpected = (((FilterMediator) thenMediatorList.get(0)).getChildThenMediatorList()
+                    .getFirstMediator() instanceof LogMediator) && asExpected;
+            asExpected = (((FilterMediator) thenMediatorList.get(0)).getChildOtherwiseMediatorList()
+                    .getFirstMediator() instanceof LogMediator) && asExpected;
+
+            // expand the filter mediator inside else block
+            asExpected = (((FilterMediator) elseMediatorList.get(0)).getChildThenMediatorList()
+                    .getFirstMediator() instanceof LogMediator) && asExpected;
+            asExpected = (((FilterMediator) elseMediatorList.get(0)).getChildOtherwiseMediatorList()
+                    .getFirstMediator() instanceof LogMediator) && asExpected;
+        } else {
+            asExpected = false;
+        }
+
+        // remove the created Integration
+        IntegrationConfigRegistry.getInstance().removeIntegrationConfig(sampleIntegration);
+
+        Assert.assertTrue("Object generation according to the provided NEL configuration has failed.", asExpected);
+    }
+
+    private boolean deployArtifacts(String iFlowResource) {
+        Artifact testArtifact = new Artifact(new File(getClass().getResource(iFlowResource).getFile()));
+        IFlowDeployer iFlowDeployer = new IFlowDeployer();
+        try {
+            iFlowDeployer.deploy(testArtifact);
+        } catch (CarbonDeploymentException e) {
+            return false;
```

Comment:
may be its not required but any reason why you are not printing at least a log when this error happens ?